### PR TITLE
stransport memory management improvements

### PR DESCRIPTION
**Pull request by pks-t**
Some fixes to memory management I've encountered when reasoning about #3885. While these fixes probably won't fix the problems encountered in this ticket, I think they are a worthwhile improvement never the less.

### CODE REVIEW

**Comment by pks-t** [comment]
Embarassing typo... should probably wait until I've got a OS X VM at hand to test changes. Meh

File: src/stransport_stream.c [hunk 1]
```diff
@@ -230,7 +230,8 @@ static void stransport_free(git_stream *stream)
 	stransport_stream *st = (stransport_stream *) stream;
 
 	git_stream_free(st->io);
-	CFRelease(st->ctx);
+	if (st->ctx)
+		CFRelease(st->ctx);
```
Comment (carlosmn) [inline src/stransport_stream.c]: When would this be `NULL`? If we fail to create the context, we error out on the constructor itself.

Comment (pks-t) [inline src/stransport_stream.c]: Agreed. The only reasoning behind this was as a safeguard. Anyway, might as well just drop it if you deem this to be unnecessary. The other fix is still valid, I guess.

Comment (carlosmn) [inline src/stransport_stream.c]: It's either this or the other block. With this patch we avoid calling the free function when it's `NULL` so it's either one or the other.

Comment (pks-t) [inline src/stransport_stream.c]: Okay, I've dropped this patch. Note though that previously we haven't been calling the free function either, as at the point of time when calling it we didn't yet initialize the `free` function pointer.

**Pull request merged**